### PR TITLE
Added via-waypoints 

### PR DESCRIPTION
**Pull request by VysotskiVadim**
Added class that represent `via_waypoints` from [Directions API](https://docs.mapbox.com/api/navigation/directions/#retrieve-directions)

### CODE REVIEW

**Review by Guardiola31337 — COMMENTED** [review]
Overall is looking good @VysotskiVadim 

Left some minor comments.

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RouteLeg.java [hunk 1]
```diff
@@ -157,6 +168,14 @@ public static RouteLeg fromJson(String json) {
   @AutoValue.Builder
   public abstract static class Builder {
 
+    /**
+     * A list of silent waypoints which were used to request a route.
+     *
+     * @param viaWaypoints a list of silent waypoints
+     * @return this builder for chaining options together
+     */
```
[the remark this answers is not in the document]
Comment (VysotskiVadim) [reply services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RouteLeg.java:177]: added 👍

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/SilentWaypoint.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+import com.google.gson.Gson;
+import com.google.gson.TypeAdapter;
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * Represents an information about silent waypoint in the leg index.
+ * See {@link RouteLeg#viaWaypoints()} for more details.
+ */
+@AutoValue
+public abstract class SilentWaypoint extends DirectionsJsonObject {
```
[the remark this answers is not in the document]
Comment (VysotskiVadim) [reply services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/SilentWaypoint.java:14]: added 👍

File: services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/DirectionsResponseTest.java [hunk 3]
```diff
@@ -109,4 +112,26 @@ public void fromJson_deserializesOptions() throws Exception {
     assertNotNull(routes.get(0).routeOptions());
     assertNotNull(routes.get(1).routeOptions());
   }
+
+  @Test
+  public void fromJson_deserializeWiaWaypoints() throws IOException {
```
[the remark this answers is not in the document]
Comment (VysotskiVadim) [reply services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/DirectionsResponseTest.java:117]: added 👍

File: services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/DirectionsResponseTest.java [hunk 4]
```diff
@@ -109,4 +112,26 @@ public void fromJson_deserializesOptions() throws Exception {
...
+    assertNotNull(viaWaypoints);
+    assertEquals(1, viaWaypoints.size());
+    SilentWaypoint waypoint = viaWaypoints.get(0);
+    assertEquals(1, waypoint.waypointIndex());
+    assertEquals(616.839, waypoint.distanceFromStart(), 0.001);
+    assertEquals(58, waypoint.geometryIndex());
+  }
+
+  @Test
+  public void testFromToJsonForRouteWithSilentWaypoints() throws IOException {
```
[the remark this answers is not in the document]
Comment (VysotskiVadim) [reply services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/DirectionsResponseTest.java]: renamed

**Review by Guardiola31337 — APPROVED** [review]
Thanks for addressing the feedback @VysotskiVadim

**Pull request merged**